drop-transfer/src/check.rs

```diff
@@ -7,55 +7,84 @@ use tokio_util::sync::CancellationToken;
 use crate::{protocol, service::State, tasks::AliveGuard, IncomingTransfer, Transfer};
 
 pub(crate) fn spawn(
+    refresh_trigger: tokio::sync::watch::Receiver<()>,
     state: Arc<State>,
     xfer: Arc<IncomingTransfer>,
     logger: Logger,
     guard: AliveGuard,
     stop: CancellationToken,
 ) -> tokio::task::JoinHandle<()> {
     let id = xfer.id();
-    let job = run(state, xfer, logger.clone());
+    let mut refresh_trigger = refresh_trigger;
 
     tokio::spawn(async move {
         let _guard = guard;
+        loop {
+            let job = run(state.clone(), xfer.clone(), logger.clone());
 
-        tokio::select! {
-            biased;
+            tokio::select! {
+                biased;
 
-            _ = stop.cancelled() => {
-                debug!(logger, "Check job stop for transfer: {id}");
-            },
-            _ = job => (),
+                _ = stop.cancelled() => {
+                    debug!(logger, "Check job stop: {id}");
+                    break;
+                },
+                _ = job => {
+                    debug!(logger, "Check job done: {id}");
+                    break;
+                },
+                _ = refresh_trigger.changed() => {
+                    debug!(logger, "Check job refresh: {id}");
+                }
+            }
```

Comment:
I think this is incorrect. You're running the job while at the same time waiting for the channel.
This should be waiting and _then_ running the lob

Reply:
@LukasPukenis I think that's the root of the bug you found. The job is running and is suddenly killed when the channel receives a thing, thus breaking the connection abruptly.

Reply:
My understanding of tokio::select! was lacking, you're totally correct on this one